File: Scadex.Model/Dtos/Cabinet/Queries/CabinetDetailDto.cs
using Scadex.Core.Model;

namespace Scadex.Model.Dtos.Cabinet.Queries;

public class CabinetDetailDto : IDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public Guid CompanyId { get; set; }
    public string CompanyName { get; set; } = null!;
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }
    public string? LocationDescription { get; set; }
    public string? GsmIp { get; set; }
    public string? NetworkIp { get; set; }
    public int? DeviceStatusId { get; set; }
    public string? DeviceStatusName { get; set; }
    public string? CreatedBy { get; set; }
    public string? UpdatedBy { get; set; }
    public DateTime? CreateDateUtc { get; set; }
    public DateTime? UpdateDateUtc { get; set; }
    public bool IsActive { get; set; }
}

File: Scadex.WebUI/src/models/cabinet/queries/cabinetDetailDto.ts
/** Ayna: Scadex.Model/Dtos/Cabinet/Queries/CabinetDetailDto.cs */
import type { DeviceStatus } from '@/models/enums';

export interface CabinetDetailDto {
  id: string;
  name: string;
  companyId: string;
  companyName: string;
  latitude: number | null;
  longitude: number | null;
  locationDescription: string | null;
  gsmIp: string | null;
  networkIp: string | null;
  /**
   * Null = kabin hiç telemetri almadı. 0 DEĞİL.
   * Alan bir zamanlar non-nullable'dı ve NULL 0'a düşüyordu; 0 =
   * `DeviceStatus.Offline` olduğu için "bilinmiyor" durumu "ÇEVRİMDIŞI" diye
   * raporlanıyordu. Bunu `docs/api-contract/samples/CabinetDetailDto.NoTelemetry.json`
   * golden'ı koruyor.
   */
  deviceStatusId: DeviceStatus | null;
  deviceStatusName: string | null;
  createdBy: string | null;
  updatedBy: string | null;
  createDateUtc: string | null;
  updateDateUtc: string | null;
  isActive: boolean;
}
